apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -227,6 +227,10 @@ type CassandraDatacenterTemplate struct {
 	// deploying Reaper in this datacenter.
 	// +optional
 	Reaper *reaperapi.ReaperDatacenterTemplate `json:"reaper,omitempty"`
+	// MgmtAPIHeap defines the amount of memory (in java format - i.e. an int followed by a G or M) devoted to the management
+	// api heap.
+	// +optional
+	MgmtAPIHeap string `json:"mgmtAPIHeap,omitempty"`
```

Comment:
This looks very much out of place at this level. Even JvmOptions would be a better place since it quite directly affects the JVM properties in the Cassandra container.

Reply:
I don't think it should be in JvmOptions because that refers to Cassandra's JVM options and I think we should be clear that this is management API heap related, not Cassandra heap related. 

I could put it [here](https://github.com/Miles-Garnsey/k8ssandra-operator/blob/1dfe518f9add98a327825f7aee8cbb668aa54e17/apis/k8ssandra/v1alpha1/k8ssandracluster_types.go#L250) wdyt?

Reply:
I agree that it should not be part
Of JvMOptions since it does not apply to Cassandra itself.

Reply:
Counterargument to my proposal above [here](https://github.com/Miles-Garnsey/k8ssandra-operator/blob/1dfe518f9add98a327825f7aee8cbb668aa54e17/apis/k8ssandra/v1alpha1/k8ssandracluster_types.go#L150). If `SystemLoggerResources` is set in the `CassandraClusterTemplate` is it better to also set the management API resources in the same place/at the same level?

We should also consider that ideally at some point it would be good to separate the Cassandra and management API processes into two containers.

Reply:
So you want the same field to be available in both `CassandraClusterTemplate` and in `CassandraDatacenterTemplate`? I currently have it in `CassandraDatacenterTemplate` only.

Reply:
Yes, it should be in both places.